spec/support/http_handling_shared.rb

```diff
@@ -148,6 +148,18 @@
         expect(sockets_used.uniq.length).to eq(1)
       end
 
+      context "on a mixed state" do
+        it "re-opens the connection" do
+          first_socket = client.get("#{server.endpoint}/socket/1").body.to_s
```

Comment:
Given the context of it being a body response I think it's fine personally, response isn't really any clearer. I did `first/last_socket_id` instead as that's more clear.